RealmSwift-swift2.0/Realm.swift

```diff
@@ -493,7 +493,13 @@ public final class Realm {
     */
     @warn_unused_result(message="You must hold on to the NotificationToken returned from addNotificationBlock")
     public func addNotificationBlock(block: NotificationBlock) -> NotificationToken {
-        return rlmRealm.addNotificationBlock(rlmNotificationBlockFromNotificationBlock(block))
+        return rlmRealm.addNotificationBlock { rlmNotification, _ in
+            if rlmNotification == RLMRealmDidChangeNotification {
```

Comment:
The reason it's slower is that converts the NSString to a swift String (allocating several new objects for some reason and copying the string data) and then does a string comparison against the raw values, while this is just a message send to `isEqual:` which returns true from the pointer comparison fast-path.